### PR TITLE
Chore: Introduce queue upkeep

### PR DESCRIPTION
**Pull request by vchan**
This PR introduces queue specific upkeep tasks. For the Postgres queue, this helps decouple opening the queue's connection pool and various upkeep tasks like the dequeue timer and enqueue listener.

### CODE REVIEW

File: tests/test_queue.py [hunk 1]
```diff
@@ -176,7 +176,7 @@ async def test_retry(self) -> None:
 
     async def test_retry_delay(self) -> None:
         # Let's first verify how things work without a retry delay
-        worker = Worker(self.queue, functions=functions, dequeue_timeout=0.01)
+        worker = Worker(self.queue, functions=functions, dequeue_timeout=0.1)
```
Comment (vchan) [inline tests/test_queue.py:179]: The low timeout was causing this test to be flaky since the Postgres queue isn't as fast as the Redis one.

File: tests/test_queue.py [hunk 2]
```diff
@@ -440,12 +440,14 @@ async def asyncTearDown(self) -> None:
         await super().asyncTearDown()
         await teardown_postgres()
 
+    @unittest.skip("Not implemented")
```
Comment (vchan) [inline tests/test_queue.py:443]: Skipping with the decorator also skips setUp and tearDown.

File: tests/helpers.py [hunk 3]
```diff
@@ -24,12 +25,18 @@ async def create_postgres_queue(**kwargs: t.Any) -> PostgresQueue:
         ),
     )
     await queue.connect()
+    await queue.upkeep()
+    await asyncio.sleep(0.1)  # Give some time for the tasks to start
```
Comment (vchan) [inline tests/helpers.py:29]: Without this, some tests were flaky because tearDown would be called before some upkeep tasks could start.

File: saq/worker.py [hunk 4]
```diff
@@ -341,6 +340,13 @@ def start(
     loop = asyncio.new_event_loop()
     worker = Worker(**settings_obj)
 
+    async def worker_start() -> None:
```
Comment (tobymao) [inline saq/worker.py:343]: why did you move this? isn't worker.stop already called on the finally of def start?

File: saq/queue/postgres.py [hunk 5]
```diff
@@ -110,31 +110,34 @@ def __init__(
...
                 )
-        # Reserve a connection for dequeue and advisory locks
-        self.connection = await self.pool.getconn()
 
-        self.tasks.append(asyncio.create_task(self.wait_for_job()))
-        self.tasks.append(asyncio.create_task(self.listen_for_enqueues()))
+    async def upkeep(self) -> set[asyncio.Task[None]]:
+        await self.init_db()
+
+        self.tasks.add(asyncio.create_task(self.wait_for_job()))
```
Comment (tobymao) [inline saq/queue/postgres.py:126]: it doesn't look like queue.tasks is used anymore, can we remove it? or do we actually need to use it?

File: tests/helpers.py [hunk 6]
```diff
@@ -24,12 +25,18 @@ async def create_postgres_queue(**kwargs: t.Any) -> PostgresQueue:
...
     return queue
 
 
 async def cleanup_queue(queue: Queue) -> None:
     if isinstance(queue, RedisQueue):
         await queue.redis.flushdb()
+    for task in queue.tasks:
```
Comment (tobymao) [inline tests/helpers.py:36]: i see queue.tasks is only used here? that seems like a smell

File: saq/worker.py (hunk 4 above)
Comment (vchan) [reply saq/worker.py:343]: I wanted to decouple queue connect/disconnect from worker start/stop. That way, if there are multiple workers sharing the same queue, stopping one won't disconnect the queue for the others. This happens in the tests, for example.

File: saq/queue/postgres.py (hunk 5 above)
Comment (vchan) [reply saq/queue/postgres.py:126]: I gave Queue its own stop() method.

**Review by tobymao — APPROVED** [review]
